test/test_helper.rb

```diff
@@ -169,3 +166,9 @@ def ssh_key_credential_from_agent
     end
   end
 end
+
+# Automatically clean up created fixture repos after the whole suite
+# there are race conditions on Windows where the original test run
+# did not release handles to the temporary directories.
+# waiting until all tests finish works successfully
+Minitest.after_run { Rugged::TestCase::FixtureRepo.teardown }
```

Comment:
The largest issue I kept running into with these tests is something about the Windows libgit2 was leaving open file handles to the temporary directories created. It appears the handles close when the test suite finishes. Therefore I think the best thing to do is clean up everything at the end of the suite instead of inline.